app/code/community/Apruve/ApruvePayment/Model/Api/Rest/Order.php

```diff
@@ -110,30 +110,47 @@ protected function _updateOrderId($apruveOrderId, $order)
      */
     public function updateOrder($apruveOrderId, $order)
     {
-        $lineItems = [];
+        $result = null;
+        $lineItems = $this->_getLineItems($order);
+
         // get discount line item
-        if(($discountItem = $this->_getDiscountItem($order))) {
+        if (($discountItem = $this->_getDiscountItem($order))) {
             $lineItems[] = $discountItem;
         }
+        $corporateAccount = Mage::getModel('apruvepayment/api_rest_account');
+        $corporateAccount->getCorporateAccount($order->getCustomerEmail());
+        $shopper_id = $corporateAccount->getShopperId($order->getCustomerEmail());
+        $payment_term = $corporateAccount->getPaymentTerm();
 
-        $data = json_encode(array(
-            'order' => array(
-                'merchant_order_id' => $order->getIncrementId(),
-                'amount_cents'      => $this->convertPrice($order->getBaseGrandTotal()),
-                'shipping_cents'    => $this->convertPrice($order->getBaseShippingAmount()),
-                'tax_cents'         => $this->convertPrice($order->getBaseTaxAmount()),
-                'invoice_on_create' => 'false',
-                'order_items'       => $lineItems
-            )
-        ));
-
-        $curlOptions = [];
-        $curlOptions[CURLOPT_POSTFIELDS] = $data;
-
-        $result = $this->execCurlRequest($this->_getUpdateOrderUrl($apruveOrderId), 'PUT', $curlOptions);
-        if($result['success'] == true) {
-            Mage::helper('apruvepayment')->logException('Order updated successfully...');
-            $this->_updateOrderId($apruveOrderId, $order);
+        if ($shopper_id) {
```

Comment:
Ok, I added code in the getShopperId method which will throw an exception if a shopper's email address is not found.